src/broker/broker.c

```diff
@@ -271,6 +271,13 @@ int main (int argc, char *argv[])
     }
     flux_watcher_start (ctx.w_internal);
 
+    /* Set early logger configuration so that flux_log(3) et al can
+     * be used before the rank is known, reactor is running, etc..
+     */
+    flux_log_set_redirect (ctx.h, log_early, ctx.attrs);
+    flux_log_set_hostname (ctx.h, "?"); // in lieu of rank
```

Comment:
On line 240 above `hostname` is already set, why not use that? (I may be missing some subtlety here though, apologies if so).